src/main/kotlin/solve/utils/nodes/DragAndDropListCell.kt

```diff
@@ -0,0 +1,112 @@
+package solve.utils.nodes
+
+import javafx.geometry.Pos
+import javafx.scene.Node
+import javafx.scene.control.ContentDisplay
+import javafx.scene.control.ListCell
+import javafx.scene.image.Image
+import javafx.scene.input.ClipboardContent
+import javafx.scene.input.DataFormat
+import javafx.scene.input.DragEvent
+import javafx.scene.input.TransferMode
+
+abstract class DragAndDropListCell<T> : ListCell<T>() {
```

Comment:
Why it is not used in catalogue?

Reply:
In the catalog, drag events were tracked at the ```ListView``` level, not at the ```ListCell``` level, because in the catalog we were not interested in swaping different elements, but only in elements selection. So in the catalogue case we did not need information about index of which cell contains this or that element